PracticeGit/intlist.c

```diff
@@ -0,0 +1,107 @@
+//----------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+//------------------------------------------------------------------------------------------------
+//   intlist.c
+//   Program on Linked List functions.
+//--------------------------------------------------------------------------------------------
+#include <malloc.h>
+#include "intlist.h"
+
+// Initialize the head pointer of the linked list to NULL.
+struct node* head = NULL;
+
+struct node* CreateList (int initialData) {
+   struct node* head = (struct node*)malloc (sizeof (struct node));
+   if (head == NULL) return ERROR_MEM_ALLOC;
+   head->data = initialData;
+   head->next = NULL;
+   return head;
+}
+
+int Append (struct node* head, int newData) {
+   struct node* current = head;
+   while (current->next != NULL) current = current->next;
+   struct node* newNode = (struct node*)malloc (sizeof (struct node));
+   if (newNode == NULL) return ERROR_MEM_ALLOC;
+   newNode->data = newData;
+   newNode->next = NULL;
+   if (head == NULL) return newNode;  // If list is empty, make newNode the head
+   current->next = newNode;
+   return SUCCESS; 
+}
+
+int Insert (struct node* head, int index, int newData) {
+   struct node* newNode = (struct node*)malloc (sizeof (struct node));
+   if (newNode == NULL) return ERROR_MEM_ALLOC;
+   struct node* current = head;
+   for (int i = 0; current != NULL && i < index - 1; i++) current = current->next;  // Traverse to the node before the specified index
+   if (current == NULL) return ERROR_INDEX_INVALID;
+   newNode->data = newData;

```

Comment:
newNode can be a Node and you can say newNode = {newData, Current->next}; Simpliflies your code.